plugins/outputs/azure_monitor/azure_monitor.go

```diff
@@ -323,12 +324,13 @@ func (a *AzureMonitor) send(body []byte) error {
 	// refresh the token if needed.
 	req, err = autorest.CreatePreparer(a.auth.WithAuthorization()).Prepare(req)
 	if err != nil {
-		return fmt.Errorf("unable to fetch authentication credentials: %v", err)
+		return fmt.Errorf("unable to fetch authentication credentials: %w", err)
 	}
 
 	resp, err := a.client.Do(req)
 	if err != nil {
-		if err.(*url.Error).Unwrap() == context.DeadlineExceeded {
+		var urlErr *url.Error
+		if errors.As(err, &urlErr) && errors.Is(urlErr, context.DeadlineExceeded) {
```

Comment:
I think this can be
```suggestion
		if errors.Is(urlErr, context.DeadlineExceeded) {
```
as the `errors.Is` function will unwrap until the target error-type is found...

Reply:
Shouldn't it be `errors.Is(err, context.DeadlineExceeded)` in that case? As `urlErr` will always be `nil`?

Reply:
I would say:
`if errors.Is(err, context.DeadlineExceeded) {`